pages/docs/support/rest-api.mdx

```diff
@@ -0,0 +1,189 @@
+
+#### Code Config Method
+
+You can also generate an API key object with the [`grouparoo generate` command]. To do this, run
+
+```
+grouparoo generate apiKey {{yourAPIKeyId}}
+```
+
+This command will generate a config file in a new directory `/config/apiKeys/{{yourAPIKeyId}}.js` that looks like this:
+
+```js
+exports.default = async function buildConfig() {
+  return [
+    {
+      class: "apiKey",
+      id: "{{yourAPIKeyId}}",
+      name: "{{yourAPIKeyId}}",
+      apiKey: "API KEY HERE" // you can hard-code the API key you want to use, or source it from `process.env.PRODUCTION_API_KEY`
+      options: {
+        permissionAllRead: false, // Does this API Key have read permission to all topics?
+        permissionAllWrite: false, // Does this API Key have write permission to all topics?
+      },
+    },
+  ];
+};
+```
+
+Grouparoo will auto-generate an API key for you, but it is also possible to set it explicitly in the config file or as an env variable `PRODUCTION_API_KEY`.
+
+From here, you can update the name and permissions.
+
+### Accessing & Testing the API
+
+As Grouparoo can be self-hosted, every deployment of Grouparoo provides its own API documentation.
+The Swagger docs for your API can be found by navigating to `/swagger` in the Grouparoo UI.
+
+<ImageInBrowserFrame
+  centered
+  source="Generate an API Key"
+  alt="Generate an API Key"
+  src="/images/docs/support/rest-api/swaggerdocs.png"
+  width={820}
+  height={530}
+/>
+
+The Grouparoo API can be authorized in 2 distinct ways: as a web user (with a cookie and CSRF token), or via an API Key. By default, when visiting the `/swagger` page, you will be authenticated using your existing cookie. You can switch to using an API key by clicking "Authorize" and entering your API Key (if pasting in your API Key, be careful about whitespace).
+
+<ImageInBrowserFrame
+  centered
+  source="Generate an API Key"
+  alt="Generate an API Key"
+  src="/images/docs/support/rest-api/auths.png"
+  width={820}
+  height={521}
+/>
+
+For example, if you want to see a list of all your instance's apps, you can run:
+
+```
+# Set Variables
+GROUPAROO_HOST="http://localhost:3000"
+GROUPAROO_API_KEY="ba05b18d9dae4acbb08799e7ef238f0d"
+
+# Make Requests
+curl -X GET "$GROUPAROO_HOST/api/v1/apps?apiKey=$GROUPAROO_API_KEY"
+```
+
+You can also make post requests. Running the command above, you should have gotten back a list of all your apps. You can use an `app_id` to generate a source like so:
+
+```
+# Set More Variables
+APP_ID="demo_db"
+SOURCE_NAME="demo_source"
+TABLE_NAME="users"
+SOURCE_TYPE="postgres-table-import"
+
+# Make Request
+curl -X POST "$GROUPAROO_HOST/api/v1/source?appId=$APP_ID&name=$SOURCE_NAME&options=%7B%22table%22%3A%20%22$TABLE_NAME%22%7D&type=$SOURCE_TYPE&state=ready&apiKey=$GROUPAROO_API_KEY"
```

Comment:
This request is putting all the variables in the URL and HTML-encoding them... which makes it kind of hard to read.  how about this style:

```bash
# Set new variables for the request
APP_ID="demo_db"
SOURCE_NAME="demo_source"
TABLE_NAME="users"
SOURCE_TYPE="postgres-table-import"

# Make Request
curl \
  -X POST \
  -d "apiKey=$GROUPAROO_API_KEY" \
  -d "appId=$APP_ID" \
  -d "name=$SOURCE_NAME" \
  -d "type=$SOURCE_TYPE" \
  -d "state=ready" \
  -d "options={\"table\": \"$TABLE_NAME\"}" \
  -d "mapping={\"id\": \"user_id\"}" \
"$GROUPAROO_HOST/api/v1/source"
```

* note the trailing `\` which lets you continue the command on a second line in bash (meaning you can paste this into the CLI) 
* Your probably also want to quote your variables
* The example is missing a mapping or a bootstrap maybe?  Or, you could make the source but not put it into the ready state... then make a property... then make it ready?

Reply:
That's much nicer!  Can you add a property to a source that isn't `"ready"`?  I'll add a mapping!